### PR TITLE
feat: cool links voting system and autothread

### PR DESCRIPTION
**Pull request by luca-montaigut**
⚠️ Add COOL_LINKS_CHANNEL_ID as an env var on the server before merging

https://github.com/codinglab-io/discord-bot/assets/17296266/326e48ff-dc25-40fb-95d7-1dafec8e9799

### CODE REVIEW

File: src/cool-links-management.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    if (!result.success) throw new Error('No OG data found');
+
+    const ogSiteName = result.ogSiteName;
+    const ogTitle = result.ogTitle;
+    if (ogSiteName && ogTitle) {
+      return `${ogSiteName} - ${ogTitle}`;
+    } else if (ogSiteName) {
+      return ogSiteName;
+    } else if (ogTitle) {
+      return ogTitle;
```
Comment (potb) [inline src/cool-links-management.ts:16]: no else/return

File: src/cool-links-management.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    return;
+  }
+
+  await message.react('✅');
+  await message.react('❌');
+
+  const threadName = await getThreadNameFromOpenGraph(detectedURLs[0]);
+  await message.startThread({
+    name: threadName ?? message.content,
+    autoArchiveDuration: 4320,
```
Comment (potb) [inline src/cool-links-management.ts:40]: put it in a constant and do something like : 1 * 60 * 60 (an hour in seconds) for it to be better understood

**Review by potb — APPROVED** [review]

File: src/cool-links-management.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const urlRegex = /(((https?:\/\/)|(www\.))[^\s]+)/g;
+  const detectedURLs = message.content.match(urlRegex);
+
+  if (detectedURLs === null) {
+    await message.delete();
+    return;
+  }
+
+  await message.react('✅');
+  await message.react('❌');
```
[the remark this answers is not in the document]
Comment (potb) [reply src/cool-links-management.ts:35]: let's ship this and see later

File: src/cool-links-management.ts (hunk 1 above)
Comment (luca-montaigut) [reply src/cool-links-management.ts]: done

File: src/cool-links-management.ts (hunk 2 above)
Comment (luca-montaigut) [reply src/cool-links-management.ts]: changed for the enum ThreadAutoArchiveDuration

**Pull request merged**